packages/data/src/components/use-select/index.js

```diff
@@ -140,6 +152,10 @@ export default function useSelect( _mapSelect, deps ) {
 	} );
 
 	useIsomorphicLayoutEffect( () => {
+		if ( isWithoutMapping ) {
```

Comment:
@youknowriad `useSelect( storeKey)` and `useSelect( map )` is not the same in terms of performance because all the mapping logic is skipped. It's not the same as calling a `useSelect( map )` twice.